crates/fuel-core/src/service/genesis.rs

```diff
@@ -222,6 +224,37 @@ fn init_contracts(
                 let root = contract.root();
                 let contract_id =
                     contract.id(&salt, &root, &Contract::default_state_root());
+                let utxo_id = if let (Some(tx_id), Some(output_idx)) =
+                    (contract_config.tx_id, contract_config.output_index)
+                {
+                    UtxoId::new(tx_id, output_idx)
+                } else {
+                    UtxoId::new(
+                        // generated transaction id([0..[out_index/255]])
+                        Bytes32::try_from(
+                            (0..(Bytes32::LEN - WORD_SIZE))
+                                .map(|_| 0u8)
+                                .chain(
+                                    (generated_output_index as u64 / 255)
+                                        .to_be_bytes()
+                                        .into_iter(),
+                                )
+                                .collect_vec()
+                                .as_slice(),
+                        )
+                        .expect("Incorrect genesis transaction id byte length"),
+                        generated_output_index as u8,
+                    )
+                };
+                let tx_pointer = if let (Some(block_height), Some(tx_idx)) = (
```

Comment:
I'm not sure how we plan to use `TxPointer` in the fork, because it is hard to verify transactions from the previous version of the network. But I guess it is okay to set it=)

Maybe we need to check that the `TxPointer.block_height <= Genesis.block_height`, otherwise it could create situations when coins from the future=)

Reply:
> I'm not sure how we plan to use TxPointer in the fork, because it is hard to verify transactions from the previous version of the network. But I guess it is okay to set it=)

The txs might not be available on the hard-fork upgrade chain, but assuming we have DA for blocks they would still be useful for fraudproofs of older blocks.